### ISSUE TITLE
Incorrect namespace resolution in anonymous classes

### ISSUE THREAD

**Issue by Norbyte** [facebook/hhvm#8275]
### HHVM Version

HipHop VM 3.27.2-dev (rel)
Compiler: heads/HHVM-3.27-0-g62a30953b2ffba94f0aeb21a3ee30a71d17e2527
Repo schema: 205b12c69b07544776201d08d23d194335873a0d
hh-9157583bf868e6c8c54d781d7584e81321c9f64f-3.27.2-dev Fri Jul 20 17:22:22 2018

### Operating System and Version

CentOS Linux release 7.5.1804 (Core)

### Standalone code, or other way to reproduce the problem

```
<?php

namespace NS2 {

    interface ArgInterface {}

    interface AnonInterface
    {
        public function fun(ArgInterface $arg);
    }

}

namespace NS1 {

    use NS2\AnonInterface;
    use NS2\ArgInterface;

    $anon = new class implements AnonInterface {
        public function fun(ArgInterface $arg)
        {
        }
    };

}
```

It seems that the newly created anonymous class is mistakenly placed into the "root" namespace instead of NS1, as using either the fully qualified name in the function definition (i.e. ` public function fun(\NS2\ArgInterface $arg)`), or a name relative to the root namespace (i.e. ` public function fun(NS2\ArgInterface $arg)`) makes the code work.
PHP 7 always puts the anonymous classes into the namespace they were declared in, so here it should be in NS1, not in the root namespace.

### Actual result

```
Fatal error: Uncaught Error: Declaration of class@anonymous$NS1\;4::fun() must be compatible with that of NS2\AnonInterface::fun() in /mnt/data/home/n/hh.php:7
Stack trace:
#0 {main}
```

### Expected result

No error

**Comment by fredemmott** [comment]
cc @oulgen , looks like https://github.com/facebook/hhvm/commit/4e716038235665a59e3506863d628f56c60ec31d didn't fully address this

**Comment by fredemmott** [comment]
There is some ongoing work to fix this in D9179686 (FB-only, sorry) , but having hopefully unrelated test issues to fix this at the moment.
